### PR TITLE
Make fal flow work with @ selector

### PR DESCRIPTION
**Pull request by mederka**

### CODE REVIEW

File: integration_tests/projects/001_flow_run_with_selectors/models/schema.yml [hunk 1]
```diff
@@ -7,6 +7,18 @@ models:
...
+  - name: empty_model
+
+  - name: empty_model_2
+    meta:
+      fal:
+        scripts:
+          before:
+            - fal_scripts/before_2.py
+          after:
+            - fal_scripts/after_2.py
```
Comment (chamini2) [inline integration_tests/projects/001_flow_run_with_selectors/models/schema.yml:20]: why different after_2 and before_2? Arent `after.py` and `before.py` enough?

**Review by chamini2 — COMMENTED** [review]
In the integration tests you have the current nodes
image
and test `@model_b` which runs
- b
- a
- ab


I am curious about this example:

image


and the same `@model_b` test. Should this run

Option 1:
- e
- d
- b
- a
- ab

Option 2:
- e
- d
- c
- b
- a
- ab

Option 3:
- e
- b
- a
- ab

**Comment by mederka** [comment]
> In the integration tests you have the current nodes image and test `@model_b` which runs
> 
>     * b
> 
>     * a
> 
>     * ab
> 
> 
> I am curious about this example:
> 
> image
> 
> and the same `@model_b` test. Should this run
> 
> Option 1:
> 
>     * d
> 
>     * b
> 
>     * a
> 
>     * ab
> 
> 
> Option 2:
> 
>     * d
> 
>     * c
> 
>     * b
> 
>     * a
> 
>     * ab
> 
> 
> Option 3:
> 
>     * b
> 
>     * a
> 
>     * ab

Option 1 for dbt, and option 3 for fal flow. I'm fixing this.

File: src/fal/cli/selectors.py [hunk 2]
```diff
@@ -76,6 +76,25 @@ def _filter_node_ids(unique_ids, fal_dbt, selected_nodes, nodeGraph) -> List[str
...
+
+
+def _get_children_with_parents(node_id, nodeGraph) -> List[str]:
+    output = []
+    children = list(nodeGraph.get_descendants(node_id))
+    output.extend(children)
+    all_parents = []
+    for child in children:
+        parents = list(nodeGraph.get_ancestors(child))
+        all_parents.extend(parents)
```
Comment (turbo1912) [inline src/fal/cli/selectors.py:92]: implement this one with reduce

Comment (mederka) [reply src/fal/cli/selectors.py:92]: is this:
```
reduce(
  lambda x: list.extend(list(nodeGraph.get_ancestors(x))),
  children,
  all_parents,
)
```
more readable that this:
```
for child in children:
  parents = list(nodeGraph.get_ancestors(child))
  all_parents.extend(parents)
```
?

🤷

Comment (chamini2) [reply src/fal/cli/selectors.py:92]: what about
```py
parentss = map(lambda child: nodeGraph.get_ancestors(child), children)
all_parents = reduce(lambda x, y: x + y, parentss, all_parents)
```

Comment (chamini2) [reply src/fal/cli/selectors.py]: still, that for is readable, just practicing

**Review by chamini2 — APPROVED** [review]

**Pull request merged**